sklearn/tests/test_discriminant_analysis.py

```diff
@@ -223,6 +224,57 @@ def test_lda_scaling():
                      'using covariance: %s' % solver)
 
 
+def test_lda_store_covariance():
+    # Test for slover 'lsqr' and 'eigen'
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+        assert_true(hasattr(clf.fit(X, y), '_covariance_'))
+
+        # Test the actual attribute:
+        clf = LinearDiscriminantAnalysis(solver=solver,
+                                         store_covariance=True).fit(X6, y6)
+        assert_true(hasattr(clf, '_covariance_'))
+
+        assert_array_almost_equal(
+            clf._covariance_[0],
+            np.array([0.422222, 0.088889])
+        )
+
+        assert_array_almost_equal(
+            clf._covariance_[1],
+            np.array([0.088889, 0.533333])
+        )
+
+    # Test for slover svd, the default is to not set the covariances_ attribute
+    clf = LinearDiscriminantAnalysis(solver='svd').fit(X, y)
+    assert_true(not hasattr(clf, '_covariance_'))
+
+    # Test the actual attribute:
+    clf = LinearDiscriminantAnalysis(solver=solver,
+                                     store_covariance=True).fit(X6, y6)
+    assert_true(hasattr(clf, '_covariance_'))
+
+    assert_array_almost_equal(
+        clf._covariance_[0],
+        np.array([0.422222, 0.088889])
+    )
+
+    assert_array_almost_equal(
+        clf._covariance_[1],
+        np.array([0.088889, 0.533333])
+    )
+
+
+def test_lda_deprecation():
+    for solver in ('lsqr', 'eigen'):
+        clf = LinearDiscriminantAnalysis(solver=solver)
+
+        # Test the deprecation
+        assert_warns_message(DeprecationWarning, "from version 0.21 "
+                             "'_covariance_' will be stored only if "
+                             "'store_covariance' is True",
+                             clf.fit, X, y)
```

Comment:
Thanks for your review! I'm not sure is this means I need a deprecation in `__getattr__` instead of now with property?
To my understanding we need `covariance_` as a "private value" but not sure which way to achieve this?